src/insights.ts

```diff
@@ -32,104 +35,111 @@ function flush(insights: SnippetAlgoliaInsights) {
-        throw new Error("init must be called first")
+      throw new Error("init must be called first");
     }
 
-    const initAction = flushedActions.shift()
-    this.init.apply(this, initAction?.args)
-    this.initalized = true;
+    const initAction = flushedActions.shift();
+    this.init.apply(this, initAction?.args);
+    this.initialized = true;
 
     flushedActions.forEach(({ methodName, args }) => {
       this[methodName].apply(this, args);
     });
   }
 
-  public init(opts: {applicationId: string, apiKey: string, region?: "us" | "de" } & UserTokenOptions) {
-    this.userToken = new UserToken({anonmyousId: opts.anonmyousId, userToken: opts.userToken})
+  public init(
+    opts: {
+      applicationId: string;
+      apiKey: string;
+      region?: "us" | "de";
+    } & UserTokenOptions
+  ) {
+    this.userToken = new UserToken({
+      anonmyousId: opts.anonmyousId,
+      userToken: opts.userToken
+    });
 
     this.beacon = new InsightsApiBeaconClient({
-        applicationId: opts.applicationId,
-        apiKey: opts.apiKey,
-        region: opts.region,
+      applicationId: opts.applicationId,
+      apiKey: opts.apiKey,
+      region: opts.region
     });
 
     // Flush and purge any existing events sitting in localStorage.
-    this.beacon.flushAndPurgeEvents()
+    this.beacon.flushAndPurgeEvents();
   }
 
   private userToken: UserToken;
   public setUserToken(userToken) {
-    this.userToken.setUserToken(userToken)
-    this.emitter.emit("userToken:changed", userToken)
+    this.userToken.setUserToken(userToken);
+    this.emitter.emit("userToken:changed", userToken);
   }
 
   public on(type: string, handler: EventEmitterCallback) {
-    this.emitter.on(type, handler)
+    this.emitter.on(type, handler);
   }
 
   private addAlgoliaAgent(agent: string) {
-    this.beacon?.addAlgoliaAgent(agent)
+    this.beacon?.addAlgoliaAgent(agent);
   }
 
-  public sendEvents(events: Array<Omit<InsightsApiEvent, "userToken" | "timestamp">>) {
-    events.forEach(event => this.sendEvent(event))
+  public sendEvents(
+    events: Omit<InsightsApiEvent, "userToken" | "timestamp">[]
+  ) {
+    events.forEach((event) => this.sendEvent(event));
   }
 
   private sendEvent(event: Omit<InsightsApiEvent, "userToken" | "timestamp">) {
-    if (!this.beacon) {
-        // TODO: log something saying you need to call init
-        // or throw an error idk.
-        return;
-    }
-
-    if (!this.userToken) {
-        // TODO: log something saying you need to call init
-        // or throw an error idk.
-        return;
+    if (!this.beacon || !this.userToken) {
+      throw new Error(
+        "Before calling any other method, you need to initialize the library by calling the 'init' function with appId and apiKey parameters"
+      );
     }
 
     this.beacon.send({
-        timestamp: Date.now(),
-        userToken: this.userToken.getUserToken(),
+      timestamp: Date.now(),
+      userToken: this.userToken.getUserToken() ?? "",
```

Comment:
If this method returns undefined I would just return early from this function. An empty `userToken` will result in an API error.

Reply:
Threw an error, but lmk if just returning is better